### ISSUE TITLE
Only allow issues to be created by contributors or people who have created pull requests

### ISSUE THREAD

**Issue by bobintetley** [dear-github/dear-github#293]
I get a lot of time-wasting issues created by people want support with an installation instead of raising genuine software bugs. 

I'd love it if there was a way that issue creation could be limited only to contributors or people who have created pull requests (ie. developers).

**Comment by wtarreau** [comment]
I have exactly the same issue and am considering giving up using github issues for this exact reason. 100% of our issues in the past were user support requests, making it impossible to use issues for development.

**Comment by cyberw** [comment]
I would love a feature for this as well. More than half of our issues are due to people not reading the documentation. I'm not sure the best solution is just blocking tickets from non-contributers alltogether is the best solution, but there should be some way.

Maybe delaying the tickets for non-contributors would be a useful compromise, so that those looking for support will use alternative ways (documentation, stackoverflow, chat, whatever other channels the project prefers) if they want a fast response rather than dev attention.

**Comment by delocalizer** [comment]
I think many of the nuisance issues (and issuers) might go away if you had something like a mana or karma feature - spend it when you create an issue as a non-contributor, replenish it by creating a PR.

**Comment by wtarreau** [comment]
> I think many of the nuisance issues (and issuers) might go away if you had something like a mana or karma feature - spend it when you create an issue as a non-contributor, replenish it by creating a PR.

No, not that way. 100% of our PRs are already pure crap and I'd love to be able to disable them as well, as they are the same nuisance as issues when they come from people who don't know what they're doing. A more correct variant of your proposal would be to say that the karma is spent when opening an issue or a PR, and replenished when the issue is confirmed or the PR merged.

**Comment by delocalizer** [comment]
> > I think many of the nuisance issues (and issuers) might go away if you had something like a mana or karma feature - spend it when you create an issue as a non-contributor, replenish it by creating a PR.
> 
> No, not that way. 100% of our PRs are already pure crap and I'd love to be able to disable them as well, as they are the same nuisance as issues when they come from people who don't know what they're doing. A more correct variant of your proposal would be to say that the karma is spent when opening an issue or a PR, and replenished when the issue is confirmed or the PR merged.

100% eh, that sounds grim ;)

Let's just say there's room for more and less hardcore variants on the same idea. We could be pretty generous because it's only a small set of outliers that burn most of the time.

**Comment by EwoutH** [comment]
I would like this. For example, Signal (https://github.com/signalapp) got totally overwhelmed with questions that should have been on their forums. Locking it for collaborators and/or contributors only could be really useful in some cases.

**Comment by danielkcz** [comment]
You can at least configure how the new issue page looks like and give people some instruction. For example, at MobX we have it like this with some instructions, directing questions to "discussions" and including useful links to docs as well. It partially helps as people have to make a conscious choice. You can also block blank issues completely so have templates with additional info.

https://github.com/mobxjs/mobx/issues/new/choose

**Comment by bobintetley** [comment]
We do the same thing, we have a template all in caps that says "ARE YOU A DEVELOPER? NO? THEN DO NOT FILE ISSUES - REQUEST SUPPORT HERE INSTEAD". 

The new option to limit interactions with a repo to existing contributors is good, but it's too broad - I only ever wanted to limit interactions to issues. If I turn that option on, it prevents people who haven't contributed before from contributing by making a pull request.

**Comment by vi** [comment]
@bobintetley Maybe instead of the message inside template there should be a form to choose templates that also includes external links to other discussion venues. [Like this one, for example](https://github.com/spacejam/sled/issues/new/choose).

**Comment by EwoutH** [comment]
> You can at least configure how the new issue page looks like and give people some instruction.

@FredyC Thanks a lot for this! They're now implemented at both [Signal-Android](https://github.com/signalapp/Signal-Android/issues/new/choose) and [Signal-iOS](https://github.com/signalapp/Signal-iOS/issues/new/choose), it looks great and hopefully it redirects users to the place they can be best helped. Thanks!

**Comment by Taitava** [comment]
I'd say now that GitHub has had Discussions for a while already, it would be nice to disable new issue creation from non-developers completely.

**Comment by DaveDavenport** [comment]
An option to limit users to first  report on the discussion forum before filing an issue would be useful.
Or can only create issues when they have atleast one thread on discussion 'upgraded' to an issue.

I have guidelines, templates and everything setup and a most issues (esp. recently) just ignore everything. (e.g. just reply  no everywhere).

At the moment I am seriously considering closing issue tracker as it is not very useful and compared to big projects I hardly get issues I don't  want to imagine how it is for them.

**Comment by mtrezza** [comment]
GitHub Issues should have a simple form builder that requires people to check boxes and fill out fields in order to submit an issue.

**Comment by DaveDavenport** [comment]
@mtrezza  that kinda exits, you can make check-boxes and everything.. but this is pointless as users just fill in dummy data (I had one say 'no' on every information query). We need a way to moderate who can create an issue.  I think the  discussion  forum gateway is a good start.

**Comment by JacobReynolds** [comment]
While we wait for this feature to implemented, I created a [Github Action](https://github.com/marketplace/actions/faceless-issues) that helps increase the quality of issue submissions. I'll take a look at adding functionality for the specific items mentioned above!

**Comment by Taitava** [comment]
## They just keep coming...

I have set up an issue template that clearly instructs to create new discussions instead of new issues: https://github.com/Taitava/obsidian-shellcommands/issues/new/choose

Even though most people notices this and follows the instructions, some people still create new issues. I guess it's because if you leave out the `/choose` part from the new issue URL, you will be presented a new issue form which doesn't show my instructions at all: https://github.com/Taitava/obsidian-shellcommands/issues/new

It's a bit frustrating that even though I have set up an issue template, it's so easy for users to accidentally circumvent it and miss the instructions. What's the point of issue templates, if they are not enforced in all issue creation routes?

**Comment by vi** [comment]
@Taitava Github Issues may be no the best medium for maintainer-only things. Maybe you can just unsubscribe from issues and use some notification bot for valid issues only? Or maybe a bot that auto-converts issues to discussions.

> Debugging and testing. For suspected bugs or weird things.
> Ideas. For feature requests.

Typically, in most projects both of those categories of user reports are handled by Issues, not by Discussions. So it is easy to see why users keep on creating issues even though guidelines recommend against it.

Maybe issue labels (which are filterable and are typically only available for maintainers) can help? Consider unlabeled issues to be equivalent to discussions, just with more steps.

Finally, Issues may be disabled in a repository altogether, though it is viewed user-unfriendly. You can have a dedicated empty repository just for issues from maintainers. 

Maybe you want to prevent issues to game some metric, e.g. to attain "not a single issue for 1 year" badge? If so then I think it is not a good idea; issues are not bugs. Just like stars and forks, they signify attention of users.

**Comment by ceolinwill** [comment]
@Taitava In the meantime, an alternative approach to disable issue creation from non-maintainers is to allow to create issues only from the `Projects` page:

- Add `blank_issues_enabled: false` to your `config.yml` file (I see you're already doing this).
- Remove your `custom.md` file.
- That's it! You'll only be able to create issues from the `Projects` page now.

**Comment by Taitava** [comment]
Thank you @wceolin, I'll need to check this! 🙂 

Edit 2024-04-14: I finally got to try this. I removed `custom.md` file. Sadly, the new issue creation form is still available via a direct link: https://github.com/Taitava/obsidian-shellcommands/issues/new . It only removed the "Get started" button from this page: https://github.com/Taitava/obsidian-shellcommands/issues/new/choose . So I assume I will still get new issues from users instead of discussions. 🙁

**Comment by Makuna** [comment]
I have been battling this exact problem, I posted a discussion on solving this, as I have been using the whole template DO NOT CREATE FOR SUPPORT and still most issues are support questions.

This was my discussion and to summarize my suggestion:  rework the ISSUES choose page, static lines to push users to discussions, and remove the BLANK issues option.

https://github.com/orgs/community/discussions/40363#discussioncomment-5427700

**Comment by jasonwilliams** [comment]
I maintain VSCode styled components, and since the latest version of VSCode broke our extension we’ve had nothing but duplicate after duplicate issue being created. It’s taking up maintainers time, and in our case, we would love a feature where we can temporarily prevent new issues being created (but still allow users to see pinned or previous issues). At least until the main problem has been resolved.

I’m sure other projects go through something similar, it’s a real problem.

https://github.com/styled-components/vscode-styled-components/issues?q=is%3Aissue+is%3Aclosed

here on this page I need to keep closing issues and telling people to go to the one already created .
![Issues · styled-componentsvscode-styled-components](https://user-images.githubusercontent.com/936006/229558514-477df725-85ba-43a4-a912-d6a66955bbf6.png)

**Comment by nefarius** [comment]
This is a real, serious issue with the platform, bigger projects regularly get raided by non-developer users when some break happens and pollute the issue tracker, causing massive moderation overhead nobody needs. This should definitely get more attention and priority, not being able to e.g. have discussions open to everyone but have the issue tracker exclusive to developers makes the repository moderation options useless IMHO.

**Comment by levino** [comment]
I think nobody posted the [workaround](https://docs.github.com/en/repositories/creating-and-managing-repositories/creating-an-issues-only-repository) so far, so I will leave it here.

And another workaround could be to have a bot/ CI script which automatically closes issues from external people and tells them to open a discussion. A bit "in your face" but what can one do?

**Comment by nefarius** [comment]
> And another workaround could be to have a bot/ CI script which automatically closes issues from external people and tells them to open a discussion. A bit "in your face" but what can one do?

I already do that, this still generates notifications that devs have to read, which wastes time spent elsewhere. The only real solution is to not being able to open "spam" issues by non-collaborator demographic in the first place. Otherwise the notifications for **real** work get polluted and need daily cleaning too.

**Comment by levino** [comment]
> > And another workaround could be to have a bot/ CI script which automatically closes issues from external people and tells them to open a discussion. A bit "in your face" but what can one do?
> 
> I already do that, this still generates notifications that devs have to read, which wastes time spent elsewhere. The only real solution is to not being able to open "spam" issues by non-collaborator demographic in the first place. Otherwise the notifications for **real** work get polluted and need daily cleaning too.

Good point. Does a clear message to people really do not help? In your case the second issue only repo would surely mitigate the problem.

**Comment by nefarius** [comment]
> Good point. Does a clear message to people really do not help? In your case the second issue only repo would surely mitigate the problem.

My experience over 10 years of FOSS communication showed me that the people who'd need to read the most can't ingest information right infront of them unless you were somehow able to deliver electric shocks to them through the Internet in the case of non-compliance. If you make something idiot-proof, the Universe simply invents better idiots. I am using issue templates, the landing page on new issues which guide to support resources and documentation, a bot that closes issues with insufficient information, and these "special" users still slip through, all safeguards are simply ignored, which is incredibly disrespectul and insulting to the valuable time of developers. The only way that worked so far is to restrict the moderation options on the repository, wich is way too coarse and also locks out users from discussions, making them entirely useless too.

**EDIT:** yes, I am thinking more and more of a separate issue repository, however I already run and offer dedicated support resources, so this seems redundant in my case.

**Comment by levino** [comment]
FWIW, I totally support the OPs idea. Should be implemented asap if you ask me. Just thought I would leave a workaround here, to be used until this no brainer has been fixed. Probably even for myself when I google the matter again in a year :D

**Comment by nefarius** [comment]
How is this still not addressed after all this time? Am considering disabling issue trackers and discussions altogether and move it off-platform where you can actually control these very basic and necessary patterns.

**Comment by AngeloGiacco** [comment]
this feels very necessary if you are using an AI agents that are based off github issues eg claude code. Otherwise it makes it trivial for someone to drain your API Key by polluting the issues

**Comment by sloraris** [comment]
Another +1 for this. I have projects I'd like to be able to display publicly while using the issue tracker with its integration with projects. I don't need anyone adding issues to some of these projects as they are very limited in collaborative scope. I can disable interactions for 6-months... why not add one more option in the dropdown for "Indefinite"?

**Comment by mk6f33** [comment]
How is this one of the most basic things still not available in 2026??!

**Comment by ufukty** [comment]
I've just filed a request in the official repository for a similar feature. Please take a look.

- https://github.com/orgs/community/discussions/191479

**Comment by gbakeman** [comment]
Everyone monitoring this, see:

https://github.blog/changelog/2026-06-29-restrict-issue-creation-to-collaborators-only/
